File: 00_onlineJudge/OLQ/5/e.cpp
#include <stdio.h>

int main(){
    int s;
    scanf("%d",&s);
    for (int x = 0; x < s; x++){
        int arr[200001] = {};
        int n;
        int maxFreq = 0;
        int maxInput = 0;
        scanf("%d",&n);
        for (int i = 0; i < n; i++){
            int x;
            scanf("%d", &x);
            if (x>maxInput){
                maxInput = x;
            }
            arr[x]++;
            if (arr[x]>maxFreq){
                maxFreq = arr[x];
            }
        }
        printf("Case #%d: %d\n", x+1, maxFreq);
        int result[maxInput] = {}, counter = 0;
        for(int i = 0; i <= maxInput; i++){ 
            if (arr[i]==maxFreq){
                result[counter]=i;
                counter++;
            }
        }

        for (int i = 0; i < counter; i++){
            if (i == counter -1) printf("%d\n",result[i]);
            else printf("%d ",result[i]);
        }
    }
    return 0;
}

File: 00_onlineJudge/OLQ/5/e.py

for i in range(int(input())):
    input()
    arr = list(map(int, input().split()))
    arr.sort(reverse=True)
    max_v = arr[0]
    max_f = 0
    count = []
    for j in range(max_v + 1):
        count.append(0)
    for j in arr:
        count[j] += 1
        if count[j] > max_f:
            max_f = count[j]
    print("Case #%d: %d" %(i+1, max_f))
    result = []
    counter = 0
    for j in range(len(count)): 
        if count[j] == max_f:
            result.append(j)
            counter += 1
    s = 0
    for j in result:
        if s == counter - 1:
            print("%d" %(j))
        else :
            print("%d" %(j), end=" ")
        s += 1

    
            


            
    